src/twisted/names/client.py

```diff
@@ -439,6 +451,43 @@ def _cbLookupZone(self, result, connector):
 
 
 
+class IPv6Resolver(Resolver):
+    """
+    Selects only IPv6 resolv.conf entries.
+    """
+    def maybeAddNameserver(self, nameserver, servers):
+        resolver = (nativeString(nameserver), dns.PORT)
+        if b':' in nameserver:
```

Comment:
3.3 added https://docs.python.org/3/library/ipaddress.html which allows you to parse an IP address, returning either a `IPv4Address` or `IPv6Address` object.  For 2.7-3.2 it has been backported and put in PyPI though: <https://pypi.python.org/pypi/ipaddress>, so might make sense to have those functions use that, to offload it from twisted.